org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/util/SemanticHighlightingTokens.java

```diff
@@ -0,0 +1,237 @@
+/*******************************************************************************
+ * Copyright (c) 2018 TypeFox GmbH (http://www.typefox.io) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.eclipse.lsp4j.util;
+
+import static java.util.Collections.emptyList;
+
+import java.nio.ByteBuffer;
+import java.util.Base64;
+import java.util.List;
+
+import org.eclipse.xtext.xbase.lib.IterableExtensions;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ComparisonChain;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableList.Builder;
+import com.google.common.collect.Iterables;
+
+/**
+ * Utility class for encoding and decoding semantic highlighting tokens into a
+ * compact, {@code base64} representation.
+ */
+public abstract class SemanticHighlightingTokens {
```

Comment:
Usually I make such a class `final`  and declare a private default constructor instead of making it abstract. An abstract class looks like something that needs to be implemented in a subclass.